store/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonWriterBuilder.java

```diff
@@ -340,7 +342,13 @@ private CarbonLoadModel createLoadModel() throws IOException, InvalidLoadOptionE
       // we are still using the traditional carbon table folder structure
       persistSchemaFile(table, CarbonTablePath.getSchemaFilePath(path));
     }
-
+    if (!table.isTransactionalTable()) {
+      CarbonProperties.getInstance()
+          .addProperty(CarbonCommonConstants.ENABLE_OFFHEAP_SORT, "false");
```

Comment:
Intention was to prevent 'unsafe' property from being used by SDK user by default.  Otherwise they have to configure memory for unsafe also. So making it simple to use. 

Any way we will update the doc how SDK user can set the property by following code:

CarbonProperties.getInstance() .addProperty("property", "value");

Reply:
But this will overwrite the ENABLE_OFFHEAP_SORT is the user has set it. Need to check if this is already set. If not then set to false otherwise use what the user has set.

Reply:
@kunal642  if we are updating the doc to set the properties then no need of this code.  same PR we can use to update the doc.